java/ql/src/semmle/code/java/JDKAnnotations.qll

```diff
@@ -19,15 +19,11 @@ class SuppressWarningsAnnotation extends Annotation {
   SuppressWarningsAnnotation() { this.getType().hasQualifiedName("java.lang", "SuppressWarnings") }
 
   /** Gets the `StringLiteral` of a warning suppressed by this annotation. */
-  StringLiteral getASuppressedWarningLiteral() {
-    result = this.getAValue() or
-    result = this.getAValue().(ArrayInit).getAnInit()
-  }
+  StringLiteral getASuppressedWarningLiteral() { result = this.getAValue(_) }
 
   /** Gets the name of a warning suppressed by this annotation. */
   string getASuppressedWarning() {
-    result = this.getAValue().(StringLiteral).getLiteral() or
-    result = this.getAValue().(ArrayInit).getAnInit().(StringLiteral).getLiteral()
+    result = this.getAValue(_).(CompileTimeConstantExpr).getStringValue()
```

Comment:
Looks like a use for `getValueString`?

Reply:
`getValueString` only works for single values, it does not work for arrays, which `@SuppressWarnings#value` uses.
I thought about adding `getAValueString` and similar for array values as well, but that would probably make the `Annotation` API too bloated.

Reply:
Have added special array value predicates now (e.g. `getAStringArrayValue(string)`) because otherwise the code accessing the array elements is verbose and error-prone, sometimes erroneously restricting values to `StringLiteral` (or similar) when they could also be a more general `CompileTimeConstantExpr`.